Modules/Server/Sources/PocketCastsServer/Public/Refresh/PodcastSearchOperation.swift

```diff
@@ -57,7 +57,7 @@ class PodcastSearchOperation: Operation {
         dispatchGroup.enter()
         URLSession.shared.dataTask(with: request) { data, _, error in
             guard let data = data, error == nil else {
-                shouldRetry = false
+                shouldRetry = true
```

Comment:
@SergioEstevao This does not fix the issue because in the line below we call the completion handler:

<img src="https://github.com/Automattic/pocket-casts-ios/assets/7040243/3ec70e2d-f851-4216-87f2-a88de26caad6" width="300">

By the way, I just re-tested this again and it seems that the issue is with the `ServerPodcastManager.shared.addFromUuid` call:

<img width="1512" alt="Screenshot 2024-02-19 at 20 51 35" src="https://github.com/Automattic/pocket-casts-ios/assets/7040243/72205e7e-107e-4194-90a6-f8b36ed5fb0d">

Reply:
I removed the completion call, and I'm now looking at that call.

Reply:
Implemented a retry logic for that call so the cache can have time to return the data.